pkg/addonmanager/controllers/addonconfig/controller.go

```diff
@@ -165,36 +163,35 @@ func (c *addonConfigController) sync(ctx context.Context, syncCtx factory.SyncCo
 		return err
 	}
 
+	// if there is unsupported configs defined in spec, skip updating spec hash.
+	configCond := meta.FindStatusCondition(addon.Status.Conditions, UnsupportedConfigurationType)
```

Comment:
I think this is just a hint for the user. You could have one config supported and another unsupported, we should still update the specHash of the supported config in this case, right?

Reply:
This function is for the framework to update the spec hash for mca.spec.configs. I'm thinking the case when at time1 user defines supported config A, and the status has spec hash for config A. Then at time2, user add unsupported configB in mca.spec.configs, and modify config A, should we continue update the spec hash for config A, and let it synced to manifestwork? or do not update anything.

Reply:
I think we should honor config A in this case.

Reply:
fixed in latest commit.